test cases/failing/127 subproject object as a dependency/test.json

```diff
@@ -0,0 +1,7 @@
+{
+  "stdout": [
+    {
+      "line": "test cases/failing/127 subproject object as a dependency/meson.build:3:0: ERROR: Argument is of an unacceptable type 'SubprojectHolder'."
```

Comment:
This... should not pass. It's not what I wanted to see, there's a specific check to emit this instead:

> ERROR: Tried to use subproject object as a dependency.
> You probably wanted to use a dependency declared in it instead.
> Access it by calling get_variable() on the subproject object.

This has been broken for quite some time and shows the generic error instead. It's not related to your PR, but actually checking it correctly requires your PR, so... I'm pushing a third commit on top of yours which fixes this properly.

Reply:
Ah, okay. I see what you meant, thank you for the clarification.

Reply:
Yeah, as you can see we should have had a test for this a looooooong time ago.